gstd-meta/codegen/src/lib.rs

```diff
@@ -0,0 +1,42 @@
+// This file is part of Gear.
+
+// Copyright (C) 2021 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+#![no_std]
+
+extern crate alloc;
+extern crate proc_macro;
+
+use alloc::string::{String, ToString};
+use proc_macro::TokenStream;
+
+#[proc_macro_attribute]
+pub fn gear_data(_attr: TokenStream, item: TokenStream) -> TokenStream {
+    let derive_string = String::from("#[derive(Debug, Decode, Encode, TypeInfo)]");
```

Comment:
this is not needed (the whole `#[gear_data]` thing)

user is quite capable of putting `#[derive(Debug, Decode, Encode, TypeInfo)]` on his own